src/bytes.rs

```diff
@@ -295,12 +312,95 @@ impl From<alloc::boxed::Box<[u8]>> for Bytes<'_> {
         }
     }
 }
-#[cfg(feature = "alloc")]
+#[cfg(feature = "stabby")]
 impl From<alloc::vec::Vec<u8>> for Bytes<'_> {
     fn from(value: alloc::vec::Vec<u8>) -> Self {
         alloc::boxed::Box::<[u8]>::from(value).into()
     }
 }
+
+#[cfg(feature = "stabby")]
+unsafe extern "C" fn retain_stabby_arc_bytes(this: *const (), capacity: usize) {
+    let this: stabby::sync::ArcSlice<u8> = unsafe {
+        stabby::sync::ArcSlice::from_raw(stabby::alloc::AllocSlice {
+            start: core::mem::transmute(this),
+            end: core::mem::transmute(capacity),
+        })
+    };
+    core::mem::forget((this.clone(), this));
```

Comment:
```suggestion
    // we don't own any `Arc` in this function:
    let this = &*::core::mem::ManuallyDrop::new(this);
    // time to do the increment:
    mem::forget(this.clone());
```

which is robust to strong count overflow (_e.g._, on 32-bit archs)